### PR TITLE
feat(cli): add `garden create project/module` commands

### PR DESCRIPTION
**Pull request by edvald**
**What this PR does / why we need it**:

See the docs and CLI help texts for details. The generated configs
contain the full project/module config schema (minus deprecated fields),
so users can easily uncomment and/or remove fields depending on their
needs.

Providers are asked to suggest modules, and I implemented that for
`container`, `helm` and `terraform` modules.

This required a bit of refactoring in our docs generation code, so a
bunch of the line changes relate to that. Also needed to make all links
in field descriptions absolute, and tweak the line breaks and formatting
for readability.

### CODE REVIEW

File: garden-service/src/commands/create/create-module.ts [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    help: "Set to false to disable interactive prompts.",
+    defaultValue: true,
+  }),
+  name: new StringOption({
+    help: "Name of the module (defaults to current directory name).",
+  }),
+  type: new StringOption({
+    help: "The module type to create. Required if --interactive=false.",
+  }),
+  // TODO: add type option
```
Comment (thsig) [inline garden-service/src/commands/create/create-module.ts]: Delete comment?

File: garden-service/src/commands/create/create-project.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    // This is to avoid `prettier` messing with the string formatting...
+    const formattedIgnoreName = chalk.bold.white(".gardenignore")
+    const configFilesUrl = chalk.cyan.underline("https://docs.garden.io/guides/configuration-files")
+    const referenceUrl = chalk.cyan.underline("https://docs.garden.io/reference/config")
+
+    log.info({
+      symbol: "info",
+      msg: wordWrap(
+        dedent`
+        We recommend reviewing the generated config, uncommenting fields that you'd like to configure, and cleaning up any commented fields that you don't need to use. Also make sure to update the ${formattedIgnoreName} file with any files you'd like to exclude from the Garden projects.
```
Comment (thsig) [inline garden-service/src/commands/create/create-project.ts]: "the Garden projects" -> "the Garden project"

File: garden-service/src/config/common.ts [hunk 3]
```diff
@@ -262,13 +262,9 @@ export const joiIdentifierDescription =
 
 const moduleIncludeDescription = (extraDescription?: string) => {
   const desc = dedent`
-  Specify a list of POSIX-style paths or globs that should be regarded as the source files for this
-  module. Files that do *not* match these paths or globs are excluded when computing the version of the module,
-  when responding to filesystem watch events, and when staging builds.
+  Specify a list of POSIX-style paths or globs that should be regarded as the source files for this   module. Files that do *not* match these paths or globs are excluded when computing the version of the module, when responding to filesystem watch events, and when staging builds.
```
Comment (thsig) [inline garden-service/src/config/common.ts]: Delete extra spaces after "source files for this".

File: garden-service/src/plugin-context.ts [hunk 4]
```diff
@@ -66,7 +66,8 @@ export const pluginContextSchema = joi
     production: joi
       .boolean()
       .default(false)
-      .description("Indicate if the current environment is a production environment"),
+      .description("Indicate if the current environment is a production environment")
```
Comment (thsig) [inline garden-service/src/plugin-context.ts]: Add period after "environment".

**Comment by edvald** [comment]
@thsig all resolved. Really no other comments than typos? :P Did you try using it?

**Comment by thsig** [comment]
> @thsig all resolved. Really no other comments than typos? :P Did you try using it?

Just took it for a more proper spin now. Some comments:
* Should we allow running create project outside of a git repo? Maybe that would just be annoying (since it would presumably have to create a directory and run `git init` for you).
* Allow specifying module name on CLI when creating module? One time, I intuitively wrote `garden create module foo` without thinking.
* The create module command allows me to create several modules with the same name if I run the command in different subdirectories of project root.
  * No error is thrown if I:
    * Run `garden create module` in the project root dir and choose the name `foo` for the module;
    * Then create a dir called foo, cd to it and run `garden create module` in it and choose `foo` again as the module name.
    * I can repeat this process with different subdirectories and the module name `foo`.

I like the UI in general, this is a nice feature!

**Comment by edvald** [comment]
All good points. One by one:

- Running outside of a git repo should be fine, just hadn't thought to test that. I can see if it's trivial to add.
- There is an optional `--name` argument for the create module command. I decided against a positional parameter, because there wasn't in mind an obvious choice between making the positional arg the module _type_ or _name_. So I went for explicit. Does that make sense?
- Yeah, we only check that the given _config_ doesn't already include a module with a certain name. We do load the project (if applicable) though, so we could just as well throw if a module with the same name already exists.

**Comment by thsig** [comment]
> There is an optional --name argument for the create module command. I decided against a positional parameter, because there wasn't in mind an obvious choice between making the positional arg the module type or name. So I went for explicit. Does that make sense?

👍

> We do load the project (if applicable) though, so we could just as well throw if a module with the same name already exists.

I think throwing here would be nice (it's easier to run the command again than to find the offending module config file and replace the name later on).

**Pull request merged**